consensus/istanbul/validator/default.go

```diff
@@ -383,7 +383,10 @@ func (valSet *defaultSet) Copy() istanbul.ValidatorSet {
 	for _, v := range valSet.validators {
 		addresses = append(addresses, v.Address())
 	}
-	return NewSubSet(addresses, valSet.policy, valSet.subSize)
+
+	newValSet := NewSubSet(addresses, valSet.policy, valSet.subSize).(*defaultSet)
+	newValSet.proposer = valSet.proposer
```

Comment:
We don't need to hard copy for proposer?

Reply:
I changed the logic!